File: Leetcode/Easy Problems/1. Two Sum/dec-19-2023.js
/*
  December 19, 2023
  - Runtime Achieved: 47 ms
  - Memory Achieved: 43.3 MB
/**
 * @param {number[]} nums
 * @param {number} target
 * @return {number[]}
 */
var twoSum = function(nums, target) {
    let dict={}
    for (let i = 0; i < nums.length; i++) {
        let a = target - nums[i];
        if (a in dict) {
            return [dict[a], i];
        }
        dict[nums[i]] = i;
    }
    return null;
};

File: Leetcode/Easy Problems/1. Two Sum/dec-19-2023.py
#  December 19, 2023
#  - Runtime Achieved: 54 ms
#  - Memory Achieved: 17.7 MB 
class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        hmap = {}
        for i in range(len(nums)):
            j = target - nums[i]
            if j in hmap:
                return [i, hmap[j]]
            hmap[nums[i]] = i
